### PR TITLE
basic port to RV32M1 RI5CY

### PR DESCRIPTION
**Pull request by unixjet**
## Summary
a basic port to RV32M1 RI5CY
## Testing
the port of Nuttx has been verified on Real RV32M1-VEGA board.

### CODE REVIEW

File: arch/risc-v/include/rv32m1/rv32m1ri5cy_irq.h [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+#define RV32M1_IRQ_BPOINT        (3) /* Break Point */
+#define RV32M1_IRQ_LAMISALIGNED  (4) /* Load Address Misaligned */
+#define RV32M1_IRQ_LAFAULT       (5) /* Load Access Fault */
+#define RV32M1_IRQ_SAMISALIGNED  (6) /* Store/AMO Address Misaligned */
+#define RV32M1_IRQ_SAFAULT       (7) /* Store/AMO Access Fault */
+#define RV32M1_IRQ_ECALLU        (8) /* Environment Call from U-mode */
+                                    /* 9-10: Reserved */
+
+#define RV32M1_IRQ_ECALLM       (11) /* Environment Call from M-mode */
+                                    /* 12-15: Reserved */
```
Comment (acassis) [inline arch/risc-v/include/rv32m1/rv32m1ri5cy_irq.h:48]: Why don't you use the definition from arch/risc-v/include/rv32im/mcause.h ? It seams to be standard for RISC-V

File: arch/risc-v/src/rv32m1/hardware/rv32m1_gpio.h [hunk 2]
```diff
@@ -0,0 +1,39 @@
+/****************************************************************************
+ * arch/risc-v/src/rv32m1/hardware/rv321m_gpio.h
```
Comment (acassis) [inline arch/risc-v/src/rv32m1/hardware/rv32m1_gpio.h:2]: you missed the "1" here rv321m_gpio.h -> rv321m1_gpio.h

File: boards/risc-v/rv32m1/rv32m1-vega/src/rv32m1_buttons.c [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+static const uint32_t g_buttons[] =
+{
+  GPIO_INPUT|GPIO_INT_EDGE|GPIO_FLOAT |GPIO_PORTA|GPIO_PIN0,  /* SW2 */
+  GPIO_INPUT|GPIO_INT_EDGE|GPIO_PULLUP|GPIO_PORTE|GPIO_PIN8,  /* SW3 */
+  GPIO_INPUT|GPIO_INT_EDGE|GPIO_PULLUP|GPIO_PORTE|GPIO_PIN9,  /* SW4 */
+  GPIO_INPUT|GPIO_INT_EDGE|GPIO_PULLUP|GPIO_PORTE|GPIO_PIN12, /* SW5 */
```
Comment (acassis) [inline boards/risc-v/rv32m1/rv32m1-vega/src/rv32m1_buttons.c:51]: Please define the buttons names and just put them here, please don't define it directly here inside the array.

File: boards/risc-v/rv32m1/rv32m1-vega/src/rv32m1_buttons.c [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+ *   After board_button_initialize() has been called, board_buttons() may be
+ *   called to collect the state of all buttons.  board_buttons() returns an
+ *   8-bit bit set with each bit associated with a button.  See the
+ *   BUTTON_*_BIT  definitions in board.h for the meaning of each bit.
+ *
+ ****************************************************************************/
+
+uint32_t board_buttons(void)
+{
+  uint8_t ret = 0;
```
Comment (acassis) [inline boards/risc-v/rv32m1/rv32m1-vega/src/rv32m1_buttons.c:95]: Should be "uint32_t ret = 0;"

**Comment by acassis** [comment]
Thank you @unixjet this is a great contribution. Your code in general has very good quality, only some small mistakes.
Before submitting the PR you could check the files using this command:
./tools/checkpatch.sh -f arch/risc-v/blablabla/filename.[ch]

**Comment by xiaoxiang781216** [comment]
Or simply run:
```
./tools/checkpatch.sh -g HEAD~...HEAD
```
which will report all nxstyle warning.

**Pull request closed without merging**